### PR TITLE
WIP: provisioning flow cleanup

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] add centerChildren flag to Box2
- [x] test signup flows
- [x] test new error flows
- [x] gen paperkey page
- [x] use changeincrement
- [x] waiting handlers on promise rpcs
- [x] sagaize all of signup
- [x] can back out of signup flow
- [x] per platform default device names
- [x] actual signup call
- [x] test mobile
- [x] split prs

### CODE REVIEW

File: protocol/bin/flow.js [hunk 1]
```diff
@@ -252,7 +252,7 @@ function engineSagaGen(methodName, name, response, requestType, responseType) {
   if (!enabledCall(methodName, 'engineSaga')) {
     return ''
   }
-  return `export const ${name}RpcSaga = (params: ${requestType}, incomingCallMap: IncomingCallMapType, loading?: (loading: boolean) => Action) => Saga.call(engineSaga, ${methodName}, params, incomingCallMap, loading)`
+  return `export const ${name}RpcSaga = (p: {params: ${requestType}, incomingCallMap: IncomingCallMapType, waitingKey?: string}) => Saga.call(engineSaga, {method: ${methodName}, params: p.params, incomingCallMap: p.incomingCallMap, waitingKey: p.waitingKey})`
```
Comment (chrisnojima) [inline protocol/bin/flow.js]: pass an object w/ keys vs unnamed params

File: shared/actions/chat2/index.js [hunk 2]
```diff
@@ -56,16 +56,16 @@ const inboxRefresh = (
     return Saga.put(Chat2Gen.createMetasReceived({clearExistingMessages, clearExistingMetas, metas}))
   }
 
-  return RPCChatTypes.localGetInboxNonblockLocalRpcSaga(
-    {
+  return RPCChatTypes.localGetInboxNonblockLocalRpcSaga({
+    incomingCallMap: {'chat.1.chatUi.chatInboxUnverified': onUnverified},
```
Comment (chrisnojima) [inline shared/actions/chat2/index.js]: use an object now

File: shared/actions/chat2/index.js [hunk 3]
```diff
@@ -56,16 +56,16 @@ const inboxRefresh = (
...
+    params: {
       identifyBehavior: RPCTypes.tlfKeysTLFIdentifyBehavior.chatGui,
       maxUnbox: 0,
       query: Constants.makeInboxQuery([]),
       skipUnverified: false,
     },
-    {'chat.1.chatUi.chatInboxUnverified': onUnverified},
-    loading => Chat2Gen.createSetLoading({key: 'inboxRefresh', loading})
-  )
+    waitingKey: 'inboxRefresh',
```
Comment (chrisnojima) [inline shared/actions/chat2/index.js]: we just use a waitingkey

File: shared/actions/json/signup.json [hunk 4]
```diff
@@ -1,58 +1,64 @@
 {
   "prelude": ["import HiddenString from '../util/hidden-string'"],
   "actions": {
-    "checkInviteCode": {
```
Comment (chrisnojima) [inline shared/actions/json/signup.json]: this and actions/signup.js are basically a big port, mostly teh same but maybe hard to see in the inline diff view

File: shared/actions/json/waiting.json [hunk 5]
```diff
@@ -6,6 +6,10 @@
     },
     "decrementWaiting": {
       "key": ["string", "Array<string>"]
-    }
+    },
+    "changeWaiting": {
```
Comment (chrisnojima) [inline shared/actions/json/waiting.json]: instead of doing increment or decrement you can do change with a flag

File: shared/actions/login.js [hunk 6]
```diff
@@ -310,19 +310,9 @@ const displayAndPromptSecretSaga = onBackSaga =>
 
 const promptNewDeviceNameSaga = onBackSaga =>
   function*({existingDevices, errorMessage}) {
-    yield Saga.put(SignupGen.createSetDeviceNameError({deviceNameError: errorMessage}))
+    yield Saga.put(SignupGen.createSubmitDevicenameDoneError({devicename: '', error: errorMessage}))
```
Comment (chrisnojima) [inline shared/actions/login.js]: this reuses the signup store which is kinda bad but i don't want to touch login now

File: shared/actions/login.js [hunk 7]
```diff
@@ -310,19 +310,9 @@ const displayAndPromptSecretSaga = onBackSaga =>
...
-          {
-            props: {
-              existingDevices,
-            },
-            selected: 'setPublicName',
-          },
-        ],
-        [loginTab, 'login']
-      )
+      navigateAppend([{props: {existingDevices}, selected: 'setPublicName'}], [loginTab, 'login'])
```
Comment (chrisnojima) [inline shared/actions/login.js]: lets not spread this over 10 lines

File: shared/actions/tracker.js [hunk 8]
```diff
@@ -618,11 +618,10 @@ function _setupTrackerHandlers() {
         }
       }
 
-      const session: Session = engine().createSession(
-        _serverCallMap(dispatch, getState, onStart, onFinish),
-        null,
-        cancelHandler
-      )
+      const session: Session = engine().createSession({
```
Comment (chrisnojima) [inline shared/actions/tracker.js]: Using an object instead of params in the function call

File: shared/constants/signup.js [hunk 9]
```diff
@@ -1,23 +1,30 @@
...
 
-const initialState: State = {
-  deviceName: isMobile ? 'Mobile Device' : 'Home Computer',
-  deviceNameError: null,
-  email: null,
-  emailError: null,
-  inviteCode: null,
-  inviteCodeError: null,
-  nameError: null,
+const devicename =
```
Comment (chrisnojima) [inline shared/constants/signup.js]: lets default to platform specific names.
@malgorithms have any opinions about the verbiage?

File: shared/constants/types/signup.js [hunk 10]
```diff
@@ -1,31 +1,22 @@
...
-export type State = {
-  deviceName: ?string,
-  deviceNameError: ?string,
-  email: ?string,
-  emailError: ?Error,
-  inviteCode: ?string,
-  inviteCodeError: ?string,
-  nameError: ?Error,
-  paperkey: ?HiddenString,
+export type _State = {
```
Comment (chrisnojima) [inline shared/constants/types/signup.js]: immutable

File: shared/devices/gen-paper-key/index.desktop.js [hunk 11]
```diff
@@ -1,9 +1,18 @@
 // @flow
```
Comment (chrisnojima) [inline shared/devices/gen-paper-key/index.desktop.js]: this isn't used in the signup flow anymore so just move it over to devices. no longer relies on Container either

File: shared/engine/index.js [hunk 12]
```diff
@@ -99,13 +100,18 @@ class Engine {
...
+    Engine._dispatch(createChangeWaiting({key, increment: waiting}))
+  }
 
   constructor(dispatch: Dispatch, getState: () => TypedState) {
-    this._dispatch = dispatch
-    this._getState = getState
+    // setup some static vars
+    Engine._dispatch = dispatch
+    Engine._getState = getState
+    Session._dispatchWaitingAction = Engine.dispatchWaitingAction
```
Comment (chrisnojima) [inline shared/engine/index.js]: this is mostly so we don't have to deal with session importing engine and dealing with import cycles / flow cycles

File: shared/engine/index.js [hunk 13]
```diff
@@ -278,51 +284,45 @@ class Engine {
...
+    const sid = this._rpcOutgoing({method, params, incomingCallMap, callback})
     return new EngineChannel(channelMap, sid, configKeys)
   }
 
   // An outgoing call. ONLY called by the flow-type rpc helpers
-  _rpcOutgoing(
+  _rpcOutgoing(p: {
     method: string,
-    params: ?{
-      incomingCallMap?: any, // IncomingCallMapType, actually a mix of all the incomingcallmap types, which we don't handle yet TODO we could mix them all
```
Comment (chrisnojima) [inline shared/engine/index.js]: instead of embedding incomingCallMap in the params lets just keep it out

File: shared/engine/saga.js [hunk 14]
```diff
@@ -26,16 +25,17 @@ type CallbackWithResponse = (any, CommonResponseHandler, TypedState) => ?RS.Effe
 type CallbackNoResponse = (any, TypedState) => ?RS.Effect | ?Generator<any, any, any>
 
 // TODO could have a mechanism to ensure only one is in flight at a time. maybe by some key or something
-function* call(
+function* call(p: {
```
Comment (chrisnojima) [inline shared/engine/saga.js]: use an object instead

File: shared/engine/saga.js [hunk 15]
```diff
@@ -26,16 +25,17 @@ type CallbackWithResponse = (any, CommonResponseHandler, TypedState) => ?RS.Effe
...
-): Generator<any, any, any> {
+  waitingKey?: string,
+}): Generator<any, any, any> {
+  const {method, params, incomingCallMap, waitingKey} = p
   const engine = getEngine()
 
-  if (waitingActionCreator) {
-    yield RSE.put(waitingActionCreator(true))
+  if (waitingKey) {
+    Engine.dispatchWaitingAction(waitingKey, true)
```
Comment (chrisnojima) [inline shared/engine/saga.js]: use waitingKeys instead of full blown actions

File: shared/login/error.render.desktop.js [hunk 16]
```diff
@@ -1,9 +0,0 @@
-// @flow
```
Comment (chrisnojima) [inline shared/login/error.render.desktop.js]: collapse all these

**Comment by chrisnojima** [comment]
replaced by 4 prs

File: shared/constants/signup.js (hunk 9 above)
Comment (malgorithms) [reply shared/constants/signup.js]: nah

**Pull request closed without merging**